crates/cext/src/transpiler/target.rs

```diff
@@ -0,0 +1,777 @@
+pub unsafe extern "C" fn qk_target_set_acquire_alignment(
+    target: *mut Target,
+    acquire_alignment: u32,
+) -> ExitCode {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { mut_ptr_as_ref(target) };
+    target.acquire_alignment = acquire_alignment;
+    ExitCode::Success
+}
+
+/// @ingroup QkTarget
+/// Creates a copy of the ``Target``.
+///
+/// @param target A pointer to the ``Target`` to copy.
+///
+/// @return A pointer to the new copy of the ``Target``.
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new(5);
+///     QkTargetEntry *entry = qk_target_entry_new(QkGate_CX);
+///     uint32_t qargs[2] = {0, 1};
+///     qk_target_entry_add_property(entry, qargs, 2, 0.0, 0.1);
+///     qk_target_add_instruction(target, entry);
+///
+///     QkTarget *copied = qk_target_copy(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_copy(target: *mut Target) -> *mut Target {
+    // SAFETY: Per documentation, the pointer is non-null and aligned.
+    let target = unsafe { const_ptr_as_ref(target) };
+
+    Box::into_raw(target.clone().into())
+}
+
+/// @ingroup QkTarget
+/// Free the ``Target``.
+///
+/// @param target A pointer to the ``Target`` to free.
+///
+/// # Example
+///     
+///     QkTarget *target = qk_target_new(5);
+///     qk_target_free(target);
+///
+/// # Safety
+///
+/// Behavior is undefined if ``target`` is not a valid, non-null pointer to a ``QkTarget``.
+#[no_mangle]
+#[cfg(feature = "cbinding")]
+pub unsafe extern "C" fn qk_target_free(target: *mut Target) {
+    if !target.is_null() {
+        if !target.is_aligned() {
+            panic!("Attempted to free a non-aligned pointer.")
+        }
+
+        // SAFETY: We have verified the pointer is non-null and aligned, so it should be
+        // readable by Box.
+        unsafe {
+            let _ = Box::from_raw(target);
+        }
+    }
+}
+
+/// Represents the mapping between qargs and ``InstructionProperties``
+#[derive(Debug)]
+pub struct TargetEntry {
+    operation: StandardGate,
+    params: Option<SmallVec<[Param; 3]>>,
+    map: IndexMap<Qargs, Option<InstructionProperties>, ahash::RandomState>,
+}
+
+impl TargetEntry {
+    pub fn new(operation: StandardGate) -> Self {
+        Self {
+            operation,
+            params: None,
+            map: Default::default(),
+        }
+    }
+
+    pub fn new_fixed(operation: StandardGate, params: SmallVec<[Param; 3]>) -> Self {
+        Self {
+            operation,
+            params: Some(params),
+            map: Default::default(),
+        }
+    }
+}
+
+/// @ingroup QkTargetEntry
+/// Creates an entry to the ``QkTarget`` based on a ``QkGate`` instance with
+/// no parameters.
+///
+/// @param operation The Standard Gate representing this entry in the target.
```

Comment:
How about this? I'm also fine keeping the original text but with ``QkGate``
```suggestion
/// @param operation The ``QkGate`` whose properties this target entry defines.
```